src/api.ts

```diff
@@ -23,3 +30,7 @@ export function setSchemaValidator(params: ValidatorContext) {
   context.validate = params.validate;
 
 }
+
+export function setDOMParserOptions(params: DOMParserContext) {
+  context.domParserOpts = params.domParserOpts || {};
```

Comment:
How about we directly create the dom instance here?

```ts

const context: Context = {
  validate: undefined,
  dom: new dom();
};

export function setDOMParserOptions(options: DOMParserContext) {
 context.dom = new dom(options);
}
```

can directly apply context.dom in the rest of the code base